src/senaite/patient/browser/patient/add2.py

```diff
@@ -0,0 +1,60 @@
+# -*- coding: utf-8 -*-
+
+from bika.lims import api
+from senaite.patient.api import get_patient_name_entry_mode
+from bika.lims.browser.analysisrequest.add2 import \
+    AnalysisRequestAddView as BaseView
+
+
+class PatientSampleAddView(BaseView):
```

Comment:
I think it would be better to use the `IGetDefaultFieldValueARAddHook` hook instead of inheriting from the base class. E.g.:

```xml
  <adapter
      factory=".addsample.PatientAddressDefaultValue"
      for="senaite.patient.interfaces.IPatient"
      provides="bika.lims.interfaces.IGetDefaultFieldValueARAddHook"
      name="senaite.patient.adapters.addsample.hook.patientaddress"/>
```

```python
@adapter(IGetDefaultFieldValueARAddHook)
class PatientAddressDefaultValue(object):

    def __init__(self, request):
        self.request = request

    def __call__(self, context):
        return context.getFormattedAddress()    
```

Reply:
For the patient case this would mean to register 7 adapters to do the same as the single override by inheritance. Therefore, I think it is cleaner and better to maintain this way.